### PR TITLE
Game controls

### PR DESCRIPTION
**Pull request by Aherbeth**

### CODE REVIEW

File: src/GameState/index.tsx [hunk 1]
```diff
@@ -148,6 +182,60 @@ export default function GameState({ navigation, route, player }: any) {
...
+        title: 'Error',
+        status: 'error',
+        description: 'Something went wrong'
+      })
+    }
+
+  }
+
+  function Controls(props: any) {
+    const directionChars = {
```
Comment (arimet) [inline src/GameState/index.tsx]: Maybe in constant file ?

**Comment by Aherbeth** [comment]
click_move_marble_mobile

**Comment by alexisjanvier** [comment]
@Aherbeth need rebase

**Comment by Aherbeth** [comment]
rebase is done :)

File: src/GameState/index.tsx [hunk 2]
```diff
@@ -206,6 +207,67 @@ export default function GameState({ navigation, route, player }: any) {
...
+          const playerForAPI = getPlayerObject(player);
+          const response = await moveMarble(
+            gameId,
+            coordinates,
+            playerForAPI,
+            direction,
+            player.token,
+          );
+
+          const game = (await response.json()) as Game;
```
Comment (alexisjanvier) [inline src/GameState/index.tsx]: You never test de status of the response ... and you dont know if you have a game in your response .... Missing to much check points ... :(

**Review by alexisjanvier — CHANGES_REQUESTED** [review]
The `src/GameState/index.tsx` file has become far too fat, so this needs to be cut out.

File: src/GameState/index.tsx [hunk 3]
```diff
@@ -206,6 +207,67 @@ export default function GameState({ navigation, route, player }: any) {
...
+
+          dispatch({ type: 'game', value: game });
```
Comment (arimet) [inline src/GameState/index.tsx]: No need for dispatch as the WS will re render

**Pull request merged**